datafusion/physical-expr/src/aggregate/utils.rs

```diff
@@ -31,3 +34,17 @@ pub fn get_accum_scalar_values_as_arrays(
         .map(|s| s.to_array_of_size(1))
         .collect::<Vec<_>>())
 }
+
+pub fn down_cast_any_ref(any: &dyn Any) -> &dyn Any {
```

Comment:
Can you  please document what this function does (with an example) given it is a new `pub` function?